cpp/non_iid_main.cpp

```diff
@@ -59,12 +64,14 @@ int main(int argc, char* argv[]){
 	unsigned long long inint;
 	char *nextOption;
 
+        string timestamp = getCurrentTimestamp();
+        string outputfilename = timestamp + ".json";
 	data.word_size = 0;
 
         initial_entropy = true;
         all_bits = true;
 
-        while ((opt = getopt(argc, argv, "icatvl:")) != -1) {
+        while ((opt = getopt(argc, argv, "icatvlo:")) != -1) {
```

Comment:
You've removed the ability for -l to have options. This looks like a typo. It should be "icatvl:o", I think.

Reply:
You need to describe what  the '-o' option does.

Reply:
It would appear that you've totally removed all non-JSON output, which is somewhat unhelpful for everyone who isn't you. :-)

Could you please try to retain the ability to use the tool using human-readable output, please? Adding an option to output JSON seems reasonable. Totally changing the output format for all users does not.

Reply:
Why would you suggest it should be `"icatvl:o"` instead of `"icatvlo:"`? I'm not fully familiar with `getopt`. I am marking `-o` as optional, so I think the `"icatvlo:"` is correct. While it was not optional, it seems like it was incorrect.

Reply:
":" indicates that the option takes an option. "-l" takes an option, but "-o" does not.

Reply:
All options are considered optional by getopt.

Reply:
Sorry, I missed that "-o" also takes an option. "-l" similarly takes an option, so the getopt string should be "icatvl:o:"